src/EditorFeatures/CSharp/BlockCommentEditing/BlockCommentEditingCommandHandler.cs

```diff
@@ -0,0 +1,177 @@
+            var currentLineStartsWithBlockCommentEndString = currentLine.StartsWith(firstNonWhitespacePosition, "*/", ignoreCase: false);
+            var currentLineStartsWithBlockCommentMiddleString = currentLine.StartsWith(firstNonWhitespacePosition, "*", ignoreCase: false);
+
+            if (!currentLineStartsWithBlockCommentStartString && !currentLineStartsWithBlockCommentMiddleString)
+            {
+                return null;
+            }
+
+            if (!IsCaretInsideBlockCommentSyntax(caretPosition))
+            {
+                return null;
+            }
+
+            if (currentLineStartsWithBlockCommentStartString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*|*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      /|*
+                    return null; // The newline inserted could break the syntax in a way that this handler cannot fix, let's leave it.
+                }
+                else
+                {
+                    //      /*|
+                    return " *" + GetPaddingOrIndentation(currentLine, caretPosition, firstNonWhitespacePosition, "/*");
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentEndString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*
+                    //      |*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      *|/
+                    return "*";
+                }
+                else
+                {
+                    //      /*
+                    //   |   */
+                    return " * ";
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentMiddleString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      *|*/
+                    return "";
+                }
+                else if (caretPosition > firstNonWhitespacePosition)
+                {
+                    //      *|
+                    return "*" + GetPaddingOrIndentation(currentLine, caretPosition, firstNonWhitespacePosition, "*");
+                }
+                else
+                {
+                    //      /*
+                    //   |   *
+                    return " * ";
+                }
+            }
+
+            return null;
+        }
+
+        private static bool BlockCommentEndsRightAfterCaret(SnapshotPoint caretPosition)
+        {
+            var snapshot = caretPosition.Snapshot;
+            return (caretPosition.Position + 2 > snapshot.Length) ? false : snapshot.GetText(caretPosition, 2) == "*/";
```

Comment:
I'm fine with that.

PS. `caretPosition + 2` bites when the overloaded operator constructs a new `SnapshotPoint` where it's position is out of the bounds of the original snapshot. So we should use either `(int)caretPosition + 2` or `caretPosition.Position + 2` here.